CMakeLists.txt

```diff
@@ -525,6 +557,22 @@ if(BASEARCH_X86_FOUND)
     )
     set(CMAKE_REQUIRED_FLAGS)
 
+    # Check whether compiler supports SSE4.2 compare string instrinics
+    check_c_source_compile_or_run(
+        "#include <immintrin.h>
+        int main(void)
+        {
+            unsigned char a[64] = { 0 };
+            unsigned char b[64] = { 0 };
+            __m128i xmm_src0, xmm_src1;
+            xmm_src0 = _mm_loadu_si128((__m128i *)(char *)a);
+            xmm_src1 = _mm_loadu_si128((__m128i *)(char *)b);
+            return _mm_cmpestri(xmm_src0, 16, xmm_src1, 16, 0);
+        }"
+        HAVE_SSE42CMPSTR_INTRIN
+    )
+    set(CMAKE_REQUIRED_FLAGS)
```

Comment:
CMAKE_REQUIRED_FLAGS was not set to enable SSE4.2, so I suspect this check will fail.
Perhaps remove the reset of the flag above the test?